pandas/tests/frame/test_constructors.py

```diff
@@ -2551,3 +2551,11 @@ def test_from_tzaware_mixed_object_array(self):
             "datetime64[ns, CET]",
         ]
         assert (res.dtypes == expected_dtypes).all()
+
+    def test_from_2d_ndarray_with_dtype(self):
```

Comment:
can you make this test more generic and add to the EA tests instead (followup ok)